### DISCUSSION TITLE
shift+up is now working

### DISCUSSION THREAD

**Discussion by antonmedv** [charmbracelet/bubbletea#727] [Q&A] [answered]
In https://github.com/antonmedv/llama for some reason shift+up/shift+down not working. (macOS, Terminal.app).

I tried to debug it but could not figure out the reason. Please help.

**Comment by bashbunni** [comment]
Hi @antonmedv can you please provide some details to reproduce? Did you install the program from source or through `brew`? What is the expected behaviour for shift+up/shift+down?

**Comment by meowgorithm** [comment]
Hey, @antonmedv! Did this ever work in `llama`? Or this this new functionality you're trying to add in? I'm looking at the Bubble Tea source code and shift+up/down should be bindable.

**Comment by meowgorithm** [comment]
Also, for what it's worth, I was able to bind to shift up/down okay with the following:

```go
package main

import (
        "fmt"
        "os"

        tea "github.com/charmbracelet/bubbletea"
)

type model string

func (m model) Init() tea.Cmd {
        return nil
}

func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
        switch msg := msg.(type) {
        case tea.KeyMsg:
                switch s := msg.String(); s {
                case "q", "esc", "ctrl+c":
                        return m, tea.Quit
                case "shift+up", "shift+down":
                        m = model(s)
                }
        }
        return m, nil
}

func (m model) View() string {
        if m == "" {
                return "Press shift+up or shift+down. Q to quit."
        }
        return "Caught " + string(m)
}

func main() {
        if _, err := tea.NewProgram(model("")).Run(); err != nil {
                fmt.Println("error while running program:", err)
                os.Exit(1)
        }
}
```

**Comment by antonmedv** [comment]
@meowgorithm thanks for looking into this issue. I tried your example, but shift+up/down now working.

shift-left/right works as expected. 

Here is a video showing this:


https://user-images.githubusercontent.com/141232/230016159-40640cd2-5ff7-4a62-ab3e-bc1ddf2ff94c.mov

**Comment by antonmedv** [comment]
I'm using a standard Terminal.app on macOS. 

```
module test

go 1.20

require github.com/charmbracelet/bubbletea v0.23.2

require (
	github.com/aymanbagabas/go-osc52 v1.2.1 // indirect
	github.com/containerd/console v1.0.3 // indirect
	github.com/lucasb-eyer/go-colorful v1.2.0 // indirect
	github.com/mattn/go-isatty v0.0.17 // indirect
	github.com/mattn/go-localereader v0.0.1 // indirect
	github.com/mattn/go-runewidth v0.0.14 // indirect
	github.com/muesli/ansi v0.0.0-20211018074035-2e021307bc4b // indirect
	github.com/muesli/cancelreader v0.2.2 // indirect
	github.com/muesli/reflow v0.3.0 // indirect
	github.com/muesli/termenv v0.14.0 // indirect
	github.com/rivo/uniseg v0.2.0 // indirect
	golang.org/x/sync v0.1.0 // indirect
	golang.org/x/sys v0.0.0-20220811171246-fbc7d0a398ab // indirect
	golang.org/x/term v0.0.0-20210927222741-03fcf44c2211 // indirect
	golang.org/x/text v0.3.7 // indirect
)
```

**Comment by bashbunni** [comment]
Can confirm, it works for me in Kitty, but not in the Terminal.app on macOS

**Comment by muesli** [comment, marked as answer]
`Terminal.app` doesn't trigger events for these two combinations. I think historically they have been used for manipulating the scrollback buffer, but they've moved to different shortcuts for that by now.

iirc, you still need to manually configure `Terminal.app` to trigger these events. See: https://github.com/timothybasanov/terminal-app-function-keys

**Comment by bashbunni** [comment]
@antonmedv 
Thanks for letting us know about this! I'm sure others will encounter a similar issue with key combinations in `Terminal.app`. I'm going to convert this to a discussion so I can mark Muesli's answer as the solution. 

Please let us know if you have any more issues with it 🙏
